server/src/mm/MmParser.ts

```diff
@@ -326,7 +327,9 @@ export class MmParser extends EventEmitter {
             }
         }
     }
-    private buildLabelToStatementMap(toks: TokenReader, currentBlock?: BlockStatement) {
+
+    // public for testing, only
```

Comment:
I've opted to use the `protected`modifier for methods that need to be accessible for unit testing, but should otherwise remain implementation details not exposed publicly. The test setup then involves creating a test class that extends the base class.

The intent is to prevent callers outside of the class hierarchy (and tests) from using these methods, keeping the public API clean. This pattern is quite common across the codebase: You can find examples by searching for `protected`.

I'd appreciate it if you could adopt this pattern for this change to keep things consistent. I'm really interested in your perspective: Do you see any significant drawbacks to this protected-for-testing pattern in TypeScript?